walnut/converter/converter.go

```diff
@@ -0,0 +1,156 @@
+/*
+ * === This file is part of ALICE O² ===
+ *
+ * Copyright 2020 CERN and copyright holders of ALICE O².
+ * Author: Ayaan Zaidi <azaidi@cern.ch>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * In applying this license CERN does not waive the privileges and
+ * immunities granted to it by virtue of its status as an
+ * Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+package converter
+
+import (
+	"fmt"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+
+	"github.com/AliceO2Group/Control/common"
+	"github.com/AliceO2Group/Control/common/controlmode"
+	"github.com/AliceO2Group/Control/common/gera"
+	"github.com/AliceO2Group/Control/core/task"
+	"github.com/AliceO2Group/Control/core/task/channel"
+)
+
+// return pointer to float64
+func createFloat(x float64) *float64 {
+	return &x
+}
+
+// return pointer to bool
+func createBool(x bool) *bool {
+	return &x
+}
+
+// return pointer to string
+func createString(x string) *string {
+	return &x
+}
+
+// ExtractTaskClasses takes in a DPL Dump string and extracts
+// an array of Tasks
+func ExtractTaskClasses(DPL Dump) (tasks []*task.Class, err error) {
+
+	for index := range DPL.Workflows {
+		var channelName string
+		if index+1 == len(DPL.Workflows) {
+			channelName = DPL.Workflows[index].Name
+		} else {
+			channelName = "from_" + DPL.Workflows[index].Name + "_to_" + DPL.Workflows[index+1].Name
+		}
+
+		taskName := DPL.Workflows[index].Name
+		defaultBindChannel := channel.Inbound{
+			Channel: channel.Channel{
+				Name:        channelName,
+				Type:        channel.ChannelType("push"), // defaulting to push
+				SndBufSize:  1000,
+				RcvBufSize:  1000,
+				RateLogging: 60,
+				Transport:   channel.TransportType("shmem"),
+			},
+			Addressing: "ipc",
+		}
+
+		// Not required for Task Templates
```

Comment:
It's true it's not required, but I'd still suggest including default values for all channels in the TTs, including `connect` (i.e. outbound) ones. The target is the only bit here that cannot be set in the TT.
Not urgent, can also be a JIRA ticket.